File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    data = {}
    for t in tangerine:
        if t not in data:
            data[t] = 1
        else:
            data[t] += 1
            
    data = dict(sorted(data.items(), key=lambda x : x[1], reverse=True))
    
    for i in data:
        if k<=0:
            return answer
        k -= data[i]
        answer += 1
    return answer
            
        

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;
import java.io.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        Map<Integer, Integer> map = new HashMap<>();
        for(int t : tangerine) {
            if(!map.containsKey(t)){
                map.put(t,1);
            } else {
                map.replace(t,map.get(t)+1);
            }
        }
        
        List<Integer> keySet = new ArrayList<>(map.keySet());
        keySet.sort((e1, e2) -> map.get(e2) - map.get(e1));
        
        for(int key : keySet){
            if(k<=0) break;
            k -= map.get(key);
            answer += 1;
        }
    
        return answer;
    }
}